src/main/java/tech/sourced/siva/SivaUnpacker.java

```diff
@@ -0,0 +1,137 @@
+package tech.sourced.siva;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.List;
+import java.util.Set;
+import java.util.zip.CRC32;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This is a high-level API for extracting files
+ * from a given .siva archive.
+ * <p>
+ * Actual reading is done by SivaReader
+ *
+ * @see SivaReader#getEntry(IndexEntry)
+ */
+public class SivaUnpacker {
+    private static final Logger log = LoggerFactory.getLogger(SivaUnpacker.class);
+    private final SivaReader reader;
+    private Boolean ignorePerms = true;
```

Comment:
final?

Reply:
Do not see, how it can be `final` in this case, because java does not have default arguments and constructor `SivaUnpacker(reader, ignorePermissions)` would not be able to override it, after calling to `SivaUnpacker(reader)`.

Reply:
You can, having a main constructor with all the arguments. Then other constructors will call to these main constructor to set default params.